Onboarding Highlights Ship Review

alessandroboron opened the pull request:
Task/Issue URL: https://app.asana.com/0/1206329551987282/1208291253556033/f

**Description**:

This PR includes changes and fixes that have been addressed during the Ship Review.

Commits are mostly self contained and address each Ship Review comment, so you can review each commit separately if you prefer.
The PR also includes the copy translations.

**Steps to test this PR**:
Test Scenario `1`, `2`, and `3` of [Ship Review Testing section](https://app.asana.com/0/1142021229838617/1208350450459625/f)

**Definition of Done (Internal Only)**:

* [ ] Does this PR satisfy our [Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f)?

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `'`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 15
* [ ] iOS 16
* [ ] iOS 17

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

alessandroboron committed:
Add translations

alessandroboron committed:
Remove progress bar text that should not be translated

alessandroboron committed:
SH1/ Fix Onboarding Address bar position button colors

alessandroboron committed:
SH2/ Fix search suggestions for US country

alessandroboron committed:
Mitgate issue where no tracker dialog is shown instead of blocked trackers

alessandroboron committed:
SH5/ Show baby ducklings images

alessandroboron committed:
SH6/ Debounce redirects

alessandroboron committed:
PH1/ Lowercase "default" string

alessandroboron committed:
SH3/ Make DuckDuckGo Search bold

alessandroboron committed:
Remove fixed offset for AppIcon picker

alessandroboron committed:
Add translations

alessandroboron committed:
Fix spacing in App Icons

alessandroboron committed:
Fix swiftlint issues

brindy commented on Core/Debouncer.swift:
```diff
@@ -0,0 +1,53 @@
+            block()
+        })
+
+        RunLoop.main.add(timer, forMode: .common)
```
Nothing about this class suggests this is for the main loop.  Can that be more explicit?  Given you've put it in a common location I'd be tempted to also pass in the RunLoop to use, maybe default it to `.current` which is what I would expect as a dev tbh.

alessandroboron replied:
fair point, I will update this and give the dev ability to choose which RunLoop to use.

alessandroboron replied:
updated, I’ve also added the RunLoop.Mode as parameter

brindy commented on Core/Debouncer.swift:
```diff
@@ -0,0 +1,53 @@
+import Foundation
+
+/// A class that provides a debouncing mechanism.
+public final class Debouncer {
```
Seems old school to implement this with timers when Combine has one built in but:

1/ Then you'd need to pass in the debounce time in the constructor
2/ It's probably just as many lines of code as this

No change required.

alessandroboron replied:
I’m fine to change to use Combine under the hood if we want, but yeah probably the lines of code will be very similar.

brindy commented on DuckDuckGo/OnboardingExperiment/ProgressBarView.swift:
```diff
@@ -144,7 +144,7 @@ struct ProgressBarGradient: View {
                     stepInfo = OnboardingProgressIndicator.StepInfo(currentStep: nextStep, totalSteps: stepInfo.totalSteps)
                 }, label: {
-                    Text("Update Progress")
+                    Text(verbatim: "Update Progress")
```
Why did this change?  `verbatim` means everyone will see "Update Progress" but shouldn't it be localised?

alessandroboron replied:
Ah this is just a SwiftUI preview I built to test the behaviour of the progress bar. When I tap that button I simulate the progress bar increasing its progress. At translation time I noticed it was translating that string.

brindy reviewed (changes requested):
Added a note to the asana task with the scenarios in them.  Couple of comments here too.

alessandroboron committed:
Fix an issue where when the user dismisses a dialog because no subsequent dialog should be presented, reloading the page would show again the dialog

alessandroboron committed:
Fix an issue that caused fire dialog to show it again when clearing browsing data and navigate to the same website before tapping on end of journey CTA

alessandroboron committed:
Fix an issue that shows the initial NTP dialog if the user clear browsing data twice before tapping on the end of journey CTA

alessandroboron committed:
Fix an issue that cause showing again the Dax dialog for last visited website when opening a new tab and visit again the website

alessandroboron committed:
Address PR comments

alessandroboron committed:
Dismiss DaxDialogs when using swipe to navigate to avoid showing blocked trackers dialog on anonymous searches page

brindy reviewed (approved):
Seems to work fine now, except for the bay thing.  Am approving but maybe we'll change eBay?

brindy reviewed (approved):
~~LGTM - tested each variant starts the expected flow~~

brindy reviewed (approved):
~~LGTM - tested each variant starts the expected flow~~

alessandroboron committed:
Onboarding Highlights - Experiment Setup (#3390)

Task/Issue URL:
https://app.asana.com/0/1206329551987282/1208084960727003/f

**Description**:
Setup 3 branches experiment for Onboarding according to [Asana comment](https://app.asana.com/0/0/1208314441219794/1208151149115582/f).

alessandroboron committed:
Fix Package.resolved
